src/FSharp.Control.AsyncSeq/AsyncSeq.fs

```diff
@@ -102,6 +102,17 @@ module internal Utils =
           Async.StartWithContinuations(a, ok, err, cnc)
           Async.StartWithContinuations(b, ok, err, cnc)
 
+      /// Creates a computation which produces a tuple consiting of the value produces by the first
+      /// argument computation to complete and a handle to the other computation. The second computation
+      /// to complete is memoized.
+      static member internal chooseBoths (a:Async<'T>) (b:Async<'U>) : Async<Choice<'T * Async<'U>, 'U * Async<'T>>> =
+        Async.chooseBoth (a |> Async.map Choice1Of2) (b |> Async.map Choice2Of2)
```

Comment:
OK I'll take a look. I did a stress test comparing two different implementations of `AsyncSeq.merge` - one using a BlockingCollection to hold incoming values from two sequences and one using `Async.chooseBoth` and the latter was marginally faster, better GC footprint, etc. Overall, they were generally comparable. I'll do a stress test focusing on `Async.chooseBoth`. One change that I think makes sense is to not use `Async.StartWithContinuations` since that uses the current thread and may be overly unfair as a result. Using `Async.Start` with explicit callbacks instead.